spec/compiler/codegen/primitives_spec.cr

```diff
@@ -310,6 +310,61 @@ describe "Code gen: primitives" do
     {% end %}
   end
 
+  describe "atomicrmw" do
+    it "codegens atomicrmw with enums" do
+      run(<<-CR).to_i.should eq(3)
+        enum RMWBinOp
+          Add = 1
+        end
+
+        enum Ordering
+          SequentiallyConsistent = 7
+        end
+
+        @[Primitive(:atomicrmw)]
+        def atomicrmw(op : RMWBinOp, ptr : Int32*, val : Int32, ordering : Ordering, singlethread : Bool) : Int32
+        end
+
+        x = 1
+        atomicrmw(:add, pointerof(x), 2, :sequentially_consistent, false)
+        x
+        CR
+    end
+
+    it "codegens atomicrmw with enums" do
+      run(<<-CR).to_i.should eq(3)
+        enum RMWBinOp
+          Add = 1
+        end
+
+        enum Ordering
+          SequentiallyConsistent = 7
+        end
+
+        @[Primitive(:atomicrmw)]
+        def atomicrmw(op : RMWBinOp, ptr : Int32*, val : Int32, ordering : Ordering, singlethread : Bool) : Int32
+        end
+
+        x = 1
+        atomicrmw(RMWBinOp::Add, pointerof(x), 2, Ordering::SequentiallyConsistent, false)
+        x
+        CR
+    end
+
+    # TODO: remove after 1.5.0
```

Comment:
We probably want to have this lingering a bit longer to avoid having to resort to the very last compiler

```suggestion
    # TODO: remove once support for 1.4 is dropped
```